src/query/api/v1/handler/prometheus/handleroptions/headers.go

```diff
@@ -91,6 +97,51 @@ func AddDBResultResponseHeaders(
 		w.Header().Add(headers.FetchedSeriesCount, fmt.Sprint(meta.FetchedSeriesCount))
 	}
 
+	aggregatedSeriesCount := 0
+	unaggregatedSeriesCount := 0
+	noSamplesCount := 0
+	withSamplesCount := 0
+	for _, m := range meta.MetadataByName {
+		aggregatedSeriesCount += m.Aggregated
+		unaggregatedSeriesCount += m.Unaggregated
+		noSamplesCount += m.NoSamples
+		withSamplesCount += m.WithSamples
+	}
+
+	if aggregatedSeriesCount > 0 {
+		w.Header().Add(headers.FetchedAggregatedSeriesCount, fmt.Sprint(aggregatedSeriesCount))
+	}
+
+	if unaggregatedSeriesCount > 0 {
+		w.Header().Add(headers.FetchedUnaggregatedSeriesCount, fmt.Sprint(unaggregatedSeriesCount))
+	}
+
+	if noSamplesCount > 0 {
+		w.Header().Add(headers.FetchedSeriesNoSamplesCount, fmt.Sprint(noSamplesCount))
+	}
+
+	if withSamplesCount > 0 {
+		w.Header().Add(headers.FetchedSeriesWithSamplesCount, fmt.Sprint(withSamplesCount))
+	}
+
+	if len(meta.Namespaces) > 0 {
+		w.Header().Add(headers.NamespacesHeader, strings.Join(meta.Namespaces, ","))
+	}
+
+	if meta.FetchedResponses > 0 {
+		w.Header().Add(headers.FetchedResponsesHeader, fmt.Sprint(meta.FetchedResponses))
+	}
+
+	if meta.FetchedBytesEstimate > 0 {
+		w.Header().Add(headers.FetchedBytesEstimateHeader, fmt.Sprint(meta.FetchedBytesEstimate))
+	}
+
+	if len(meta.MetadataByName) > 0 {
+		stats := meta.TopMetadataByName(maxMetricStatsInHeader)
+		js, _ := json.Marshal(stats)
```

Comment:
Should probably log this error at least

Reply:
I went ahead and returned the error similar to WaitedAny